qiskit_nature/transformers/second_quantization/electronic/__init__.py

```diff
@@ -0,0 +1,39 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+QMolecule Transformers (:mod:`qiskit_nature.transformers.second_quantization`)
+==============================================================================
+
+.. currentmodule:: qiskit_nature.transformers.second_quantization
+
+Transformers act on a :class:`~qiskit_nature.drivers.second_quantization.QMolecule` to produce an
+altered copy of it as per the specific transformer. So for instance the
+:class:`FreezeCoreTransformer` will alter the integrals and number of particles in a way that
+freezes the core orbitals, storing an extracted energy in the QMolecule to compensate for this that
+would need to be included back into any ground state energy computation to get complete result.
+
+.. autosummary::
+   :toctree: ../stubs/
+
+   ActiveSpaceTransformer
+   FreezeCoreTransformer
+
+"""
+
+from .active_space_transformer import ActiveSpaceTransformer
+from .freeze_core_transformer import FreezeCoreTransformer
+
+__all__ = [
```

Comment:
We used to expose `BaseTransformer` here in all and in the docs - so any third party can implement their own transformer etc. Also we expose that class as the type for transformers in ElectronicProblem so I think it should be here as it was previously (at least for now while Transformers are as they are acting on QMolecule)

Reply:
Ah I see what I am missing. I did not move `BaseTransformer` into the `electronic` submodule here.. That is not necessary because it defines the interface based on `DriverResult` which is the electronic/vibrational-agnostic return type.

In short: `BaseTransformer` is exposed in `qiskit_nature.transformers.second_quantization`.

Reply:
Ok, I now see they are documented both in the parent folder as well as this sub-folder. I assume they were exposed in the parent as we do not expect, and/or will prevent, any name collisions around transformers, and this is the level from which we would normally expect users to import them. If so I would drop them from being documented in electronic folder as well - a user can of course still choose to import them from there if they want even if they don't appear in API refs docs at that level - but being available at the level above why would anyone?

Reply:
No, the reason they are also documented in the parent folder is because of their deprecated variants (currently) being located there. I will fix this to make proper sense after #283 is merged.

Reply:
Ok, I don't think we really need to document the deprecated versions though.

Reply:
I removed the unreleased code in `transformers/second_quantization`.
I kept the documentation of `transformers` but added a deprecation notice there, too. Or shall I directly remove that docstring alltogether?

Reply:
@woodsp-ibm knows better about this. For me it is fine as is.

Reply:
I know at times with Aqua we deprecate/relocated classes and at times, we had them in the docs. I guess its ok. I will note there is a [..deprecated](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-deprecated) in Sphinx too that I used in the past where you can add info like what is its replacement.

If we do this for transformers I think for consistency it should be done with drivers too - I think there we just have the new ones.